std/range/package.d

```diff
@@ -5277,24 +5278,23 @@ if ((isIntegral!(CommonType!(B, E)) || isPointer!(CommonType!(B, E)))
 
         this(Value current, Value pastLast, StepType step)
         {
-            if ((current < pastLast && step >= 0) ||
-                    (current > pastLast && step <= 0))
+            if ((current < pastLast && step > 0) ||
+                    (current > pastLast && step < 0))
             {
                 this.step = step;
                 this.current = current;
                 if (step > 0)
                 {
                     assert(unsigned((pastLast - current) / step) <= size_t.max);
-
-                    this.pastLast = pastLast - 1;
+                    // Cast below can't fail because current < pastLast
+                    this.pastLast = cast(Value) (pastLast - 1);
```

Comment:
> It's a a toss seeing that this variant has more duplication.

True, but not duplication of conditionals that have already been checked :) There is also the inferred logic in your comment (if step > 0, then current must be < lastPast due to how the big if condition is constructed) that becomes unnecessary when the actual if condition is simplified.

This seems to me like the debate about combining versions and repeating the declarations. One is more DRY, but the other makes it clear which declarations are active for a specific version.

In any case, I'm OK with the code as written as it is correct and not really more confusing than the original, it just was hard to follow.